File: src/main/java/pl/tsimur/jpatest/repository/OrganizationRepository.java
package pl.tsimur.jpatest.repository;

import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import pl.tsimur.jpatest.model.Organization;

import java.util.List;

@Repository
public interface OrganizationRepository extends CrudRepository<Organization, Integer> {

    Organization findOrganizationByNip(Integer nip);

    List<Organization> findAllOrganizationByName(String name);

    List<Organization> findAllOrganizationByAddress(String address);

    @Query(value = "SELECT org.name " +
            " FROM organization org INNER JOIN user_account u ON  org.nip = u.employer_nip WHERE u.age > 50", nativeQuery = true)
    List<String> findAllOrgNamesWithPensioners();

    @Modifying
    @Query(value = "UPDATE Organization SET address = :name WHERE nip = :nip")
    void updateOrganizationByNip(Integer nip, String name);

    @Modifying
    void deleteByNip(Integer nip);
}

File: src/main/kotlin/pl/timur/jpatest/repository/OrganizationRepository.kt
package pl.timur.jpatest.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import pl.timur.jpatest.model.Organization

@Repository
interface OrganizationRepository : JpaRepository<Organization, Int> {
    fun findOrganizationByNip(nip: Int): Organization?
    fun findAllOrganizationByName(name: String): List<Organization>
    fun findAllOrganizationByAddress(address: String): List<Organization>

    @Query(
        value = "SELECT org.name " +
                " FROM organization org INNER JOIN user_account u ON  org.nip = u.employer_nip WHERE u.age > 50",
        nativeQuery = true
    )
    fun findAllOrgNamesWithPensioners(): List<String>

    @Modifying
    @Query(value = "UPDATE Organization SET address = :name WHERE nip = :nip")
    fun updateOrganizationByNip(nip: Int, name: String)

    @Modifying
    fun deleteByNip(nip: Int)
}